src/main/scala/io/iohk/ethereum/blockchain/sync/fast/PivotBlockSelector.scala

```diff
@@ -63,10 +63,15 @@ class PivotBlockSelector(
       log.info(
         "Cannot pick pivot block. Need at least {} peers, but there are only {} which meet the criteria ({} all available at the moment). Retrying in {}",
         minPeersToChoosePivotBlock,
-        correctPeers.size,
-        peersToDownloadFrom.size,
+        eligiblePeers.size,
+        allPeers.size,
         startRetryInterval
       )
+      if (election.isEnoughVoters(minPeersToChoosePivotBlock + 1)) { // TODO when we should start blacklisting best peers?
```

Comment:
To be honest i am not sure about it. I am worried about the case like:
- we start our node, and we connect to one bootstrap node (which is deemed as truthful) 
- we find 3 just starting peers on block 1000
- we blacklist our bootstrap as it is to far a long
- we sync up to one of low block nodes on block 950

Maybe we should have similiar mechanism as besu that has few levels of checking  like: at first check if there are voters in range `best - 32` ,then in `best - 64` etc. up to lets say `128` ? It is not perferct but a little more robust to differnt network condition than our current solution.

Reply:
:heavy_plus_sign: on this.

Besu implements this strategy in [PivotBlockRetriever](https://github.com/hyperledger/besu/blob/8b6f8b6e2f8f498af71523580eb0d6e6779d06c7/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/PivotBlockRetriever.java), [FastSyncActions](https://github.com/hyperledger/besu/blob/8b6f8b6e2f8f498af71523580eb0d6e6779d06c7/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/FastSyncActions.java) and perhaps [PivotBlockConfirmer](https://github.com/hyperledger/besu/blob/8b6f8b6e2f8f498af71523580eb0d6e6779d06c7/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/PivotBlockConfirmer.java), with the [default distance](https://github.com/hyperledger/besu/blob/8b6f8b6e2f8f498af71523580eb0d6e6779d06c7/ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/SynchronizerConfiguration.java#L29) being 50.